### DISCUSSION TITLE
How do combine files but keep chapters?

### DISCUSSION THREAD

**Discussion by morpheus2n2** [yermak/AudioBookConverter#246] [Q&A] [answered]
I have Audiobooks that have been split in to separate mp3 files each one is a chapter, how do I combine all these into a single m4b while keeping the chapters? 

First time I did it I had a load of single m4b files, the next time I hit combine but all I got was a single m4b file but it had no chapters at all

**Comment by yermak** [comment]
If you select "Split by chapters" - you will have multiple m4b where each file is matching chapter to your original mp3 file.
If you select "Split by parts" - you book will be split by parts, by default 1 part, and your chapters will be there - you could test it with you ios device or with VLC.

**Reply by morpheus2n2** [reply]
So I can't combine them into one single file that's chaptered?

**Reply by yermak** [reply]
You can combine them into single chaptered file - that's **the purpose** of program.

>> had no chapters at all

How did you check it?

**Reply by morpheus2n2** [reply]
Mediainfo, what started with 17 chapters via mp3 ends up being a single chapter of 8 hours! It is not treating them the way it should
<img width="1920" alt="2021-04-08" src="https://user-images.githubusercontent.com/14253602/114029450-6a796880-9871-11eb-995a-1ce35f649f9c.png">

**Reply by yermak** [reply]
So... I have several thoughs here - assuming you use the latest version...
1. Please try another book and check if you have the same issue
2. if you could use command line - in app folder there there is program ffprobe.exe - please try to run it ffprobe.exe [resulting file]
3. it's diffucult to diagnose the issue w/o original files in place it could be that setting chapters fails for this particular fileset. 


Options are: 
- you upload source book somewhere to cloud and give me a link - mynickname @ gmail.com and i will try to reproduce. 
- You send me you log file after you convert this book - better to delete it before this. In my case it's located "C:\Users\Yermak\Application Data\AudioBookConverter-5.5.18\AudioBookConverter.log"  you should replace you username to get it.

**Reply by morpheus2n2** [reply]
Dose it with all books, I think its me doing it wrong but cos I can't find any actual info on how to do this its hard to say

**Reply by Chriss78** [reply]
I have the same issue, tryed diffrent things that book just wont go into one file with chapters. I managed to get it into one file but chapters are gone. Version is 6.2.10

**Reply by godweaver** [reply, marked as answer]
I am having the same issue. I tried both splitting by chapters and parts. Splitting by chapters prior to conversion got me multiple m4b files but not one full audiobook with multiple chapters within. Splitting by parts ended in one large m4b audiobook with no options to skip to the different chapters under "part 1" (audiobook title). I have version 6.4.2.

**Reply by yermak** [reply]
If you want a SINGLE book with MULTIPLE chapters - you need to select "Split by Parts" this will create as single book with 1 Part.
Chapters will be inside a book which you can navigate with you player, e.g. open the m4b book with VLC - this player support chapter navigation.
Might be helpful to watch video here 
https://store.steampowered.com/app/1529240/AudiobookConverter/

**Reply by cottonias** [reply]
> I am having the same issue. I tried both splitting by chapters and parts. Splitting by chapters prior to conversion got me multiple m4b files but not one full audiobook with multiple chapters within. Splitting by parts ended in one large m4b audiobook with no options to skip to the different chapters under "part 1" (audiobook title). I have version 6.4.2.

I found this comment because I was having the same issue. Turns out I was on an old version (6.2 or something) - it had worked in the past but was now outputting many files for one book. I downloaded the latest version (6.5.1), Windows Single User and it worked just fine (7 files merged into 1, with chapter markers, using "Split by" set to "parts").

https://github.com/yermak/AudioBookConverter/releases
